### PR TITLE
Refactor networking portion of the EVE-K3s integration

### PR DESCRIPTION
**Pull request by milan-zededa**
These changes are explained in [this document](https://docs.google.com/document/d/1smRfJ8xwx-FSoKQmWZVL_dlcAbw8UizUpMOYYi8BXbk/edit?usp=sharing).
Note that before opening PRs against the upstream, I would cleanup the git history, add detailed commit messages, etc.

### CODE REVIEW

File: pkg/pillar/cmd/zedrouter/cni.go [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+func (z *zedrouter) getAppByPodName(
+	podName string) (*types.AppNetworkConfig, *types.AppNetworkStatus, error) {
+	appName, appUUIDPrefix, err := kubeapi.GetAppNameFromPodName(podName)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, item := range z.pubAppNetworkStatus.GetAll() {
+		appStatus := item.(types.AppNetworkStatus)
+		if appStatus.DisplayName == appName &&
```
Comment (naiming-zededa) [inline pkg/pillar/cmd/zedrouter/cni.go:329]: Need to convert the 'DisplayName' to lower case to match in this case

**Review by naiming-zededa — COMMENTED** [review]
@milan-zededa your PR changes look good to me. and one comment on the checking and matching to App display name during the RPC call, we need normalize to all lower cases in the DisplayName. @zedi-pramodh has updated the poc branch to the latest of dec 3rd. According to him the 'poc-dec3' is working. so you probably should merge over there, instead of using this branch.

File: pkg/pillar/cmd/zedrouter/cni.go (hunk 1 above)
Comment (milan-zededa) [reply pkg/pillar/cmd/zedrouter/cni.go:329]: Done in https://github.com/naiming-zededa/eve/pull/230

**Comment by milan-zededa** [comment]
Replaced by https://github.com/naiming-zededa/eve/pull/230

**Pull request closed without merging**